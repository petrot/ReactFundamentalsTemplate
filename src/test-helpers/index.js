export * from "./local-storage.mock";
export * from "./store.mock";
export * from "./test-wrapper.mock";
