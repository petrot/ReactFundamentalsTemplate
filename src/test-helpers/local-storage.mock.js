export const mockGetItem = jest.fn();
export const mockSetItem = jest.fn();
export const mockRemoveItem = jest.fn();

export const prepareMockLocalStorage = () => {
  Object.defineProperty(window, "localStorage", {
    value: {
      getItem: (...args) => mockGetItem(...args),
      setItem: (...args) => mockSetItem(...args),
      removeItem: (...args) => mockRemoveItem(...args),
    },
  });
};
