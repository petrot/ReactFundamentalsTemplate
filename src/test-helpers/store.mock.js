export const TEST_NAME = "Test Name";

export const mockedState = {
  user: {
    isAuth: true,
    name: TEST_NAME,
    role: "admin",
    token: "mock-token",
  },
  courses: [
    {
      title: "Course 1",
      description: "Description 1",
      duration: 5,
      authors: [
        "9b87e8b8-6ba5-40fc-a439-c4e30a373d36",
        "1c972c52-3198-4098-b6f7-799b45903199",
        "072fe3fc-e751-4745-9af5-aa9eed0ea9ed",
      ],
      creationDate: "11/03/2024",
      id: "b28fd9ad-c683-4900-bf99-38a1c5748309",
    },
  ],
  authors: [
    {
      name: "author",
      id: "9b87e8b8-6ba5-40fc-a439-c4e30a373d36",
    },
    {
      name: "author2",
      id: "1c972c52-3198-4098-b6f7-799b45903199",
    },
    {
      name: "author3",
      id: "072fe3fc-e751-4745-9af5-aa9eed0ea9ed",
    },
  ],
};

export const mockedStore = {
  getState: () => mockedState,
  subscribe: jest.fn(),
  dispatch: jest.fn(),
};
