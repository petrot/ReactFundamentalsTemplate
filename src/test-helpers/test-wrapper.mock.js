import { Provider } from "react-redux";
import { MemoryRouter as Router } from "react-router-dom";
import { mockedStore } from "./store.mock";

export const TestWrapper = ({ children, store = mockedStore }) => (
  <Provider store={store}>
    <Router>{children}</Router>
  </Provider>
);
