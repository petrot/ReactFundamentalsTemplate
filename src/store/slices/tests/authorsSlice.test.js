import { mockedState } from "../../../test-helpers";
import reducer, { setAuthors, saveAuthor } from "../authorsSlice";

describe("authorsSlice", () => {
  it("should return the initial state", async () => {
    expect(reducer(undefined, {})).toEqual([]);
  });

  it("should handle a course list being added to an empty list", async () => {
    const previousState = [];

    expect(
      reducer(previousState, setAuthors(mockedState.authors))
    ).toMatchSnapshot();
  });

  it("should handle a course being added", async () => {
    const previousState = [...mockedState.authors];
    const author = {
      name: "author4",
      id: "365fe3fc-e751-4745-9af5-bf9eed0ea9de",
    };

    expect(reducer(previousState, saveAuthor(author))).toMatchSnapshot();
  });
});
