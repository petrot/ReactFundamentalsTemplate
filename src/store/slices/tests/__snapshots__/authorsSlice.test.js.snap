// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authorsSlice should handle a course being added 1`] = `
Array [
  Object {
    "id": "9b87e8b8-6ba5-40fc-a439-c4e30a373d36",
    "name": "author",
  },
  Object {
    "id": "1c972c52-3198-4098-b6f7-799b45903199",
    "name": "author2",
  },
  Object {
    "id": "072fe3fc-e751-4745-9af5-aa9eed0ea9ed",
    "name": "author3",
  },
  Object {
    "id": "365fe3fc-e751-4745-9af5-bf9eed0ea9de",
    "name": "author4",
  },
]
`;

exports[`authorsSlice should handle a course list being added to an empty list 1`] = `
Array [
  Object {
    "id": "9b87e8b8-6ba5-40fc-a439-c4e30a373d36",
    "name": "author",
  },
  Object {
    "id": "1c972c52-3198-4098-b6f7-799b45903199",
    "name": "author2",
  },
  Object {
    "id": "072fe3fc-e751-4745-9af5-aa9eed0ea9ed",
    "name": "author3",
  },
]
`;
